tests/components/blebox/test_light.py

```diff
@@ -0,0 +1,597 @@
+    assert state.state == STATE_OFF
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(entry.device_id)
+
+    assert device.name == "My wLightBox"
+    assert device.identifiers == {("blebox", "abcd0123ef5678")}
+    assert device.manufacturer == "BleBox"
+    assert device.model == "wLightBox"
+    assert device.sw_version == "1.23"
+
+
+async def test_wlightbox_update(wlightbox, hass, config):
+    """Test light updating."""
+
+    feature_mock, entity_id = wlightbox
+
+    def initial_update():
+        feature_mock.is_on = True
+        feature_mock.rgbw_hex = "fa00203A"
+        feature_mock.white_value = 0x3A
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    await async_setup_entity(hass, config, entity_id)
+
+    state = hass.states.get(entity_id)
+    assert state.attributes[ATTR_HS_COLOR] == (352.32, 100.0)
+    assert state.attributes[ATTR_WHITE_VALUE] == 0x3A
+    assert state.state == STATE_ON
+
+
+async def test_wlightbox_on_via_just_whiteness(wlightbox, hass, config):
+    """Test light on."""
+
+    feature_mock, entity_id = wlightbox
+
+    def initial_update():
+        feature_mock.is_on = False
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    await async_setup_entity(hass, config, entity_id)
+    feature_mock.async_update = AsyncMock()
+
+    state = hass.states.get(entity_id)
+    assert state.state == STATE_OFF
+
+    def turn_on(value):
+        feature_mock.is_on = True
+        assert value == "f1e2d3c7"
+        feature_mock.white_value = 0xC7  # on
+        feature_mock.rgbw_hex = "f1e2d3c7"
+
+    feature_mock.async_on = AsyncMock(side_effect=turn_on)
+
+    def apply_white(value, white):
+        assert value == "f1e2d305"
+        assert white == 0xC7
+        return "f1e2d3c7"
+
+    feature_mock.apply_white = apply_white
+
+    feature_mock.sensible_on_value = "f1e2d305"
+
+    await hass.services.async_call(
+        "light",
+        SERVICE_TURN_ON,
+        {"entity_id": entity_id, ATTR_WHITE_VALUE: 0xC7},
+        blocking=True,
+    )
+
+    state = hass.states.get(entity_id)
+    assert state.state == STATE_ON
+    assert state.attributes[ATTR_WHITE_VALUE] == 0xC7
+
+    assert state.attributes[ATTR_HS_COLOR] == color.color_RGB_to_hs(0xF1, 0xE2, 0xD3)
+
+
+async def test_wlightbox_on_via_reset_whiteness(wlightbox, hass, config):
+    """Test light on."""
+
+    feature_mock, entity_id = wlightbox
+
+    def initial_update():
+        feature_mock.is_on = False
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    await async_setup_entity(hass, config, entity_id)
+    feature_mock.async_update = AsyncMock()
+
+    state = hass.states.get(entity_id)
+    assert state.state == STATE_OFF
+
+    def turn_on(value):
+        feature_mock.is_on = True
+        feature_mock.white_value = 0x0
+        assert value == "f1e2d300"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5535891)